### PR TITLE
Fix unit tests for /apis, /internal, /pkg

### PR DESCRIPTION
**Pull request by lisguo**
*Description of changes:*
Unit test fixes, can run full suite of unit tests with `make test`

Code snippet of tests passing in packages
```
ok  	github.com/aws/amazon-cloudwatch-agent-operator/apis/v1alpha1	(cached)
ok  	github.com/aws/amazon-cloudwatch-agent-operator/internal/config	(cached)
ok  	github.com/aws/amazon-cloudwatch-agent-operator/internal/manifests/collector/adapters	(cached)
ok  	github.com/aws/amazon-cloudwatch-agent-operator/internal/manifests/manifestutils	(cached)
ok  	github.com/aws/amazon-cloudwatch-agent-operator/internal/naming	(cached)
ok  	github.com/aws/amazon-cloudwatch-agent-operator/internal/version	(cached)
ok  	github.com/aws/amazon-cloudwatch-agent-operator/internal/webhook/podmutation	(cached)
ok  	github.com/aws/amazon-cloudwatch-agent-operator/pkg/featuregate	(cached)
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -67,14 +67,6 @@ rules:
   - patch
   - update
   - watch
-- apiGroups:
```
Comment (lisguo) [inline config/rbac/role.yaml:70]: this section is unneeded as it is duplicated from line [107](https://github.com/aws/amazon-cloudwatch-agent-operator/pull/63/files#diff-6d4c4e29ec096149781536bb9118fde8242c1901df764e3eda3019ac8503d6c7R99)

File: Makefile [hunk 2]
```diff
@@ -82,10 +84,10 @@ ci: test
 # Run tests
 # setup-envtest uses KUBEBUILDER_ASSETS which points to a directory with binaries (api-server, etcd and kubectl)
 .PHONY: test
-test: generate fmt vet ensure-generate-is-noop envtest
```
Comment (lisguo) [inline Makefile:85]: `ensure-generate-is-noop ` doesn't exist in the Makefile, we don't use it

File: Makefile [hunk 3]
```diff
@@ -51,6 +51,8 @@ CONTROLLER_TOOLS_VERSION ?= v0.12.0
 ALL_SRC := $(shell find . -name '*.go' -type f | sort)
 CW_AGENT_OPERATOR_IMPORT_PATH = "github.com/aws/amazon-cloudwatch-agent-operator"
 
+GOTEST_OPTS := $(shell go list ./... | grep -v integration-tests)
```
Comment (lisguo) [inline Makefile:54]: We can ignore the integration-tests as part of `make test`

File: config/manager/kustomization.yaml [hunk 4]
```diff
@@ -4,5 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: aws/amazon-cloudwatch-agent-operator
-  newTag: latest
+  newName: aws/cloudwatch-agent-operator
+  newTag: 1.0.2
```
Comment (lisguo) [inline config/manager/kustomization.yaml:8]: This was made to be consistent with the image produced in the Makefile by `make container`

**Comment by jefchien** [comment]
Thanks for fixing the unit tests! Can you summarize the major changes you had to make?

File: internal/manifests/collector/suite_test.go [hunk 5]
```diff
@@ -51,7 +51,7 @@ func paramsWithMode(mode v1alpha1.Mode) manifests.Params {
 				UID:       instanceUID,
 			},
 			Spec: v1alpha1.AmazonCloudWatchAgentSpec{
-				Image: "ghcr.io/aws/amazon-cloudwatch-agent-operator/amazon-cloudwatch-agent-operator:0.47.0",
+				Image: "public.ecr.aws/cloudwatch-agent/cloudwatch-agent:1.300031.1b317",
```
Comment (jefchien) [inline internal/manifests/collector/suite_test.go:54]: Is there a reason we're using this specific image?

File: Makefile (hunk 3 above)
Comment (jefchien) [reply Makefile:54]: nit: Could add an `integration-test` build tag to the integration tests, so they're skipped normally.

```go
//go:build integration-test
```

**Review by jefchien — APPROVED** [review]

**Review by jefchien — APPROVED** [review]

File: apis/v1alpha1/collector_webhook.go [hunk 6]
```diff
@@ -23,9 +23,9 @@ var (
...
-// +kubebuilder:webhook:verbs=create;update,path=/validate-cloudwatch-aws-amazon-com-v1alpha1-amazoncloudwatchagent,mutating=false,failurePolicy=fail,groups=cloudwatch.aws.amazon.co,resources=amazoncloudwatchagents,versions=v1alpha1,name=vamazoncloudwatchagentcreateupdate.kb.io,sideEffects=none,admissionReviewVersions=v1
-// +kubebuilder:webhook:verbs=delete,path=/validate-cloudwatch-aws-amazon-com-v1alpha1-amazoncloudwatchagent,mutating=false,failurePolicy=ignore,groups=cloudwatch.aws.amazon.co,resources=amazoncloudwatchagents,versions=v1alpha1,name=vamazoncloudwatchagentdelete.kb.io,sideEffects=none,admissionReviewVersions=v1
+// +kubebuilder:webhook:path=/mutate-cloudwatch-aws-amazon-com-v1alpha1-amazoncloudwatchagent,mutating=true,failurePolicy=fail,groups=cloudwatch.aws.amazon.com,resources=amazoncloudwatchagents,verbs=create;update,versions=v1alpha1,name=mamazoncloudwatchagent.kb.io,sideEffects=none,admissionReviewVersions=v1
```
Comment (mitali-salvi) [inline apis/v1alpha1/collector_webhook.go:26]: Thank you for fixing this

**Review by mitali-salvi — APPROVED** [review]
